llvm/lib/Transforms/Scalar/LICM.cpp

```diff
@@ -2088,6 +2093,8 @@ bool llvm::promoteLoopAccessesToScalars(
         FoundLoadToPromote = true;
 
         Align InstAlignment = Load->getAlign();
+        LoadIsGuaranteedToExecute |=
```

Comment:
Maybe wrap this in `if (!LoadIsGuaranteedToExecute)` to save unnecessary queries.

Reply:
Done, thanks!